span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/ExcludeSpanRuleEvaluator.java

```diff
@@ -112,29 +112,45 @@ private boolean matchesRelationalSpanFilter(
         case FIELD_ENVIRONMENT_NAME:
           Optional<String> environmentMaybe =
               HttpSemanticConventionUtils.getEnvironmentForSpan(event);
-          if (environmentMaybe.isEmpty()) {
-            return false;
-          }
-          return spanFilterMatcher.matches(
-              environmentMaybe.get(),
-              relationalSpanFilterExpression.getRightOperand(),
-              relationalSpanFilterExpression.getOperator());
+          return environmentMaybe
+              .filter(
+                  environment ->
+                      spanFilterMatcher.matches(
+                          environment,
+                          relationalSpanFilterExpression.getRightOperand(),
+                          relationalSpanFilterExpression.getOperator()))
+              .isPresent();
         case FIELD_URL:
-          Optional<String> fullHttpUrlMaybe = HttpSemanticConventionUtils.getFullHttpUrl(event);
-          if (fullHttpUrlMaybe.isEmpty()) {
-            return false;
-          }
-          return spanFilterMatcher.matches(
-              fullHttpUrlMaybe.get(),
-              relationalSpanFilterExpression.getRightOperand(),
-              relationalSpanFilterExpression.getOperator());
+          Optional<String> urlMaybe = getUrl(event);
```

Comment:
`getUrl` doesn't look like it's falling back to HTTP path. Can we instead use `HttpSemanticConventionUtils.getFullHttpUrl(event).or(() ->  HttpSemanticConventionUtils.getHttpPath(event));`